src/components/shared/ButtonWithPanel/ArrowPanel.js

```diff
@@ -94,7 +96,11 @@ export class ArrowPanel extends React.PureComponent<Props, State> {
           onClick={this._onArrowPanelClick}
         >
           <div className="arrowPanelArrow" />
-          <div className="arrowPanelContent">{children}</div>
+          <div className="arrowPanelContent">
+            <ClosePanelContext.Provider value={this.close}>
```

Comment:
I wasn't sure whether I should use this context here or in `ButtonWithPanel`.
For now I found it simpler to put this here, because this component holds the logic about how the panel closes (with the animation handling especially). If we use react-transition instead in the future we may be able to simplify this component and make it fully controlled by `ButtonWithPanel` (using an `open` prop).

Reply:
This is an outdated comment ^. The idea is that a Context provides the callback to close the panel. In this PR there's nothing about that at the moment and I don't know if I'll keep it in the upcoming patch, but I still like the idea...